mu-plugins/central-hub/tests/phpunit/unit/config-store/getConfig.php

```diff
@@ -0,0 +1,81 @@
+<?php
+/**
+ *  Tests for getConfig()
+ *
+ * @package    spiralWebDb\centralHub\Tests\Unit\ConfigStore
+ * @since      1.3.0
+ * @author     Robert A. Gadon
+ * @link       https://github.com/rgadon107/cornerstone
+ * @license    GNU General Public License 2.0+
+ */
+
+namespace spiralWebDb\centralHub\Tests\Unit\ConfigStore;
+
+use function KnowTheCode\ConfigStore\getConfig;
+use function KnowTheCode\ConfigStore\_the_store;
+use spiralWebDb\Cornerstone\Tests\Unit\Test_Case;
+
+/**
+ * Class Tests_GetConfig
+ *
+ * @package spiralWebDb\centralHub\Tests\Unit\ConfigStore
+ * @group   config-store
+ */
+class Tests_GetConfig extends Test_Case {
+
+	/**
+	 * Prepares the test environment before each test.
+	 */
+	protected function setUp() {
+		parent::setUp();
+
+		require_once CENTRAL_HUB_ROOT_DIR . '/src/config-store/internals.php';
+		require_once CENTRAL_HUB_ROOT_DIR . '/src/config-store/api.php';
+	}
+
+	/**
+	 * Test should return configuration from _the_store() when store key is given.
+	 */
+	public function test_should_return_configuration_when_store_key_is_given() {
+		$config = (array) require CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/test-cpt-config.php';
+		_the_store( 'foo', $config );
+		$expected = getConfig( 'foo' );
+
+		$this->assertArrayHasKey( 'foo', $expected );
+		$this->assertSame( $expected, $config );
+
+		// Clean up _the_store.
+		_the_store( 'foo', null, true );
+	}
+
+	/**
+	 * Test should throw Exception when configuration does not exist in store.
+	 */
+	public function test_should_throw_exception_when_config_does_not_exist() {
+		$this->expectException( \Exception::class );
+		$this->expectExceptionMessage( 'Configuration for [foo] does not exist in the ConfigStore' );
+
+		getConfig( 'foo' );
+	}
+
+	/**
+	 * Test should return empty array when store key does not exist.
+	 */
+	public function test_should_return_empty_array_when_key_does_not_exist() {
+		$config = (array) require CENTRAL_HUB_ROOT_DIR . '/tests/phpunit/fixtures/test-cpt-config.php';
+		_the_store( '', $config );
+
+		$this->assertSame( [], getConfig( '' ) );
```

Comment:
@rgadon107 For this test, we are checking if what happens when the key does not exist in the store. Here are some thoughts to improve this test:

1. There’s no need to add a config with no key.  It won’t store it in the store.
2. What happens if you text for the method name as a key?

Combining these two, you could rewrite this test to:

```php
public function test_should_return_empty_array_when_key_does_not_exist() {
     $this->assertSame( [], getConfig( ‘’ ) );
     $this->assertSame( [], getConfig( ‘this_key_does_not_exist’ ) );
     $this->assertSame( [], getConfig( __METHOD__ ) );
}
```

What do notice from this test?

1.  You don’t need to handle the store.
2.  You are testing multiple keys to ensure you get what you expect.  

In other words, you’re checking more than an empty string to ensure the behavior works with different combinations.  Why?  It’s possible that the code checks for an empty string and has a different behavior with a different string.  We check that the behavior is consistent in each of those edge cases.

Does that make sense?

Reply:
@hellofromtonya This makes perfect sense. It avoids dealing directly with `_the_store()`. And it tests the expected behavior and conditions more broadly. I like the approach. I'll incorporate that in.

Reply:
@hellofromtonya The test for `'test_should_return_empty_array_when_key_does_not_exist'` only works for the first assert. The second and third asserts throw the Exception in `_the_store`. I added the second and third asserts to the method `'test_should_throw_exception_when_config_does_not_exist'` and that test passed.

Reply:
@hellofromtonya All three possible cases are now tested: 

1) Given store key & config array keyed by the store key;
2) Given store key & no config. Exception thrown.
3) Given no store key & no config. Returns empty array. 

Can this PR be approved?